src/lua/lua_task.c

```diff
@@ -2644,22 +2644,48 @@ lua_task_inject_url(lua_State *L)
 
 	if (lua_isuserdata(L, 3)) {
 		/* We also have a mime part there */
-		mpart = *((struct rspamd_mime_part **) rspamd_lua_check_udata_maybe(L,
-																			3, rspamd_mimepart_classname));
+		mpart = *((struct rspamd_mime_part **)
+							   rspamd_lua_check_udata_maybe(L,3,rspamd_mimepart_classname));
 	}
-
 	if (task && task->message && url && url->url) {
 		if (rspamd_url_set_add_or_increase(MESSAGE_FIELD(task, urls), url->url, false)) {
+			struct rspamd_mime_text_part *text_part;
+			text_part = rspamd_mempool_alloc0(task->task_pool,
```

Comment:
Why do we need a fake text part here? Please check how it's done in HTML urls extraction: https://github.com/rspamd/rspamd/blob/cd92d8bb1d412abc4aaaea57efb55145f9f49b8b/src/libserver/html/html.cxx#L749